src/osd/PrimaryLogPG.cc

```diff
@@ -769,6 +790,20 @@ class PGLSParentFilter : public PGLSFilter {
                       bufferlist& outdata) override;
 };
 
+bool PGLSNameFilter::filter(const hobject_t &obj,
+                              bufferlist& indata, bufferlist& outdata)
+{
+  // match user regex
+  if (regex != "" && !regex.empty() && regex != "no_reg") {
```

Comment:
Put that in init() function, so it doens't need to be executed for each object. Doing this on a pool with 1M objects will certainly have performance impact on that OSD.